inference-engine/tests/functional/plugin/myriad/subgraph_tests/concat_split_transpose.cpp

```diff
@@ -0,0 +1,86 @@
+        LayerTestsUtils::TargetDevice>;
+
+class Concat_Split_Transpose : public testing::WithParamInterface<Parameters>, virtual public LayerTestsUtils::LayerTestsCommon {
+protected:
+    void SetUp() override {
+        SetRefMode(LayerTestsUtils::RefMode::CONSTANT_FOLDING);
+        configuration[InferenceEngine::MYRIAD_DISABLE_CONVERT_STAGES] = CONFIG_VALUE(YES);
+        configuration[InferenceEngine::MYRIAD_DETECT_NETWORK_BATCH] = CONFIG_VALUE(NO);
+
+        const auto& dataType = std::get<0>(GetParam());
+        const auto& dataDims = std::get<1>(GetParam());
+        const auto& axis = std::get<2>(GetParam());
+        const auto& length = std::get<3>(GetParam());
+        targetDevice = std::get<4>(GetParam());
+
+        auto params = ngraph::builder::makeParams(dataType, dataDims);
+        auto paramOuts = ngraph::helpers::convert2OutputVector(
+                ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+        auto concat = std::make_shared<ngraph::opset1::Concat>(paramOuts, axis);
+
+        const auto lengthData = std::make_shared<ngraph::opset3::Constant>(ngraph::element::i64,
+                                                                            ngraph::Shape{length.size()},
+                                                                            length);
+        const auto axisData = std::make_shared<ngraph::opset3::Constant>(ngraph::element::i64,
+                                                                          ngraph::Shape{1},
+                                                                          axis);
+        auto split = std::make_shared<ngraph::opset3::VariadicSplit>(concat, axisData, lengthData);
+
+        auto permutation = std::vector<std::int64_t>(split->get_output_shape(0).size());
+        std::iota(permutation.rbegin(), permutation.rend(), 0);
+        const auto transposition = std::make_shared<ngraph::opset3::Constant>(ngraph::element::i64,
+                                                                              ngraph::Shape{split->get_output_shape(0).size()},
+                                                                              permutation);
+
+        ngraph::ResultVector results;
+        for (int i = 0; i < 2; i++) {
+            const auto transpose = std::make_shared<ngraph::opset3::Transpose>(split->output(i), transposition);
+            results.push_back(std::make_shared<ngraph::opset1::Result>(transpose));
+        }
+        function = std::make_shared<ngraph::Function>(results, params, "concat-split-transpose");
+    }
+};
+
+TEST_P(Concat_Split_Transpose, CompareWithRefs) {
+    Run();
+}
+
+std::vector<DataDims> dims = {
+        {{400, 1}, {600, 1}}
+};
+
+std::vector<std::vector<std::size_t>> length = {
+        {500, 500}
+};
+
+INSTANTIATE_TEST_CASE_P(SpecialStages, Concat_Split_Transpose,
+                        ::testing::Combine(
+                                ::testing::Values(ngraph::element::i32),
```

Comment:
Does it make sense to check other data types (ngraph::element::f16, ngraph::element::f32)?

Reply:
And more operations types as well would be nice (expand, crop, reshape ...)

Reply:
Even more, I'm not sure such a test covers much in your changes. You're changing the code which is responsible for checks whether we need to replace some special stage with copy. Those checks rely on specific attributes (e.g. strides consistency) - so variation in those attributes should take place in the test

Reply:
It is specific test-reproducer of issue with yolo-v3-pytorch. I guess no need to expand this test case.

Reply:
Otherwise it is just refactoring without changes in behavior.

Reply:
I still don't think the issue is specific to yolo-v3-pytorch. The model is just an example where it has been reproduced. Expanding testing coverage is not a refactoring. Since we didn't find the issue at the development stage, it's seems to me more than just reasonable to make sure there is no others similar issues. Could be done separately, though.

Reply:
OK, I make a ticket with tech-debt.